scanner/src/main/kotlin/storages/PostgresStorage.kt

```diff
@@ -89,40 +92,42 @@ class PostgresStorage(
      * Setup the database.
      */
     fun setupDatabase() {
-        val statement = connection.createStatement()
-        val resultSet = statement.executeQuery("SHOW client_encoding")
-        if (resultSet.next()) {
-            val clientEncoding = resultSet.getString(1)
-            if (clientEncoding != "UTF8") {
-                log.warn { "The database's client_encoding is '$clientEncoding' but should be 'UTF8'." }
-            }
-        }
+        executeStatement {
+            executeQuery("SHOW client_encoding").use { resultSet ->
```

Comment:
Commit message:

> Also make sure that all JDBC resources are explicitly closed, which
was not the case before.

Actually *that* bit *is* something I would have preferred to see in a separate preceding commit, because it's a valid fix independently of introducing a datasource, and if we ever needed to revert the datasource changes, we should not also loose the fix to close the JDBC resources.

Reply:
I was thinking about this as well. However, the problem is that almost all places that need to be adapted to close the resources have to be changed/reworked again for introducing the DataSource; especially when introducing helper functions like _execute()_ and _executeStatement()_. So this would actually mean double work

Reply:
I agree it's a *bit* of double work. For me, a cleaner Git history would be worth the effort, but I can understand your point of view and won't insist on changing it.